core/dbt/parser/sources.py

```diff
@@ -137,15 +137,13 @@ def parse_source(self, target: UnpatchedSourceDefinition) -> ParsedSourceDefinit
         tags = sorted(set(itertools.chain(source.tags, table.tags)))
 
         config = self._generate_source_config(
-            fqn=target.fqn,
+            target=target,
             rendered=True,
-            project_name=target.package_name,
         )
 
         unrendered_config = self._generate_source_config(
-            fqn=target.fqn,
+            target=target,
```

Comment:
Background context on `rendered: bool`, `UnrenderedConfigGenerator`, and `unrendered_config`, since these are pretty confusing!

This exists to limit "false positives" during state comparison, i.e. to power `state:modified` selection in Slim CI runs ("just build what's changed"): https://docs.getdbt.com/reference/node-selection/state-comparison-caveats#false-positives

The comparison happens in the `same_config` method here:
https://github.com/dbt-labs/dbt-core/blob/899b0ef2241d97d97df5270c20b67be02ec7ca61/core/dbt/contracts/graph/parsed.py#L317-L321

Idea being, it's not uncommon for users to have environment-based configuration like:
```yml
sources:
  - name: my_source
    config:
      database: "{{ 'prod_raw' if target.name == 'prod' else 'sampled_dev' }}"
      schema: "{{ env_var('DBT_SOURCE_SCHEMA') }}"
      # etc
```
The rendered versions of those configs will be different based on the values of env vars or `target` values. The thing we _actually_ want to detect changes in is the raw un-rendered Jinja expression. If that's changed, it's because someone edited the thing.

This works for configs set in `dbt_project.yml`, which are stored as unrendered Jinja expressions, and passed twice into `calculate_node_config` (once with rendering, once without).

However, we've _already_ rendered all fields in (non-project) yaml files, during initial parsing:
https://github.com/dbt-labs/dbt-core/blob/899b0ef2241d97d97df5270c20b67be02ec7ca61/core/dbt/parser/schemas.py#L633-L635

So we're passing already-rendered Jinja expressions into `calculate_node_config`, and we never get access to their unrendered forms. We'd need a subtle reordering of parsing/rendering to make it work as intended.

Related: https://github.com/dbt-labs/dbt-core/issues/2744, https://github.com/dbt-labs/dbt-core/issues/3576